MoreLinq/CountBy.cs

```diff
@@ -78,51 +78,35 @@ static partial class MoreEnumerable
 
                 void Loop(IEqualityComparer<TKey> cmp)
                 {
-                    var dic = new Dictionary<TKey, int>(cmp);
-                    var nullIndex = (int?) null;
-
-                    bool TryGetIndex(TKey key, out int i)
-                    {
-                        if (key == null)
-                        {
-                            i = nullIndex.GetValueOrDefault();
-                            return nullIndex.HasValue;
-                        }
-
-                        return dic.TryGetValue(key, out i);
-                    }
+                    var dic = new Collections.Dictionary<TKey, int>(cmp);
```

Comment:
It's actually related. This refactoring, as well as any other, was strictly done to avoid introducing uses of dammit operator (`!`) if it could be helped. I started by refactoring the local function into a normal static method, where the parameter list grew. Then I saw the same pattern and issue in `ScanBy` and decided to extract to a separate type.